bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -56,6 +56,20 @@ where
         let conn = self.builder.build_connection(Mode::Server)?;
         TlsStream::open(conn, stream).await
     }
+
+    pub async fn accept_with_application_context<S, T>(
+        &self,
+        stream: S,
+        app_context: T,
+    ) -> Result<TlsStream<S, B::Output>, Error>
+    where
+        S: AsyncRead + AsyncWrite + Unpin,
+        T: Send + Sync + 'static,
+    {
+        let mut conn = self.builder.build_connection(Mode::Server)?;
+        conn.as_mut().set_application_context(app_context);
+        TlsStream::open(conn, stream).await
+    }
```

Comment:
This seems like too specific of a method. We'd potentially need accept_with variations for all the possible different ways to build a connection.

Have you seen [connection builders](https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/connection/builder.rs#L12-L16)? Would a builder work for this use case?